tests/conftest.py

```diff
@@ -71,17 +76,68 @@ def database_url(test_db):
     ).scalar()
     assert count_landsat == count_landsat_schema
 
-    return test_db.connection.engine.url
+    return str(test_db.connection.engine.url)
+
+
+def create_tipg_app(
+    postgres_settings: PostgresSettings,
+    db_settings: DatabaseSettings,
+    sql_settings: CustomSQLSettings,
+) -> FastAPI:
+    """Create TiPG Application."""
+
+    from tipg.db import close_db_connection, connect_to_db, register_collection_catalog
+    from tipg.factory import Endpoints
+
+    app = FastAPI(
+        title="TiPg: OGC Features and Tiles API",
+        openapi_url="/api",
+        docs_url="/api.html",
+    )
+    ogc_api = Endpoints(title="TiPg: OGC Features and Tiles API")
+    app.include_router(ogc_api.router)
+
+    @app.on_event("startup")
+    async def startup_event() -> None:
+        """Connect to database on startup."""
+        await connect_to_db(
+            app,
+            settings=postgres_settings,
+            schemas=db_settings.user_schemas,
+            user_sql_files=sql_settings.sql_files,
+        )
+        await register_collection_catalog(
+            app,
+            schemas=db_settings.schemas,
+            exclude_schemas=db_settings.exclude_schemas,
+            tables=db_settings.tables,
+            exclude_tables=db_settings.exclude_tables,
+            function_schemas=db_settings.function_schemas,
+            exclude_function_schemas=db_settings.exclude_function_schemas,
+            functions=db_settings.functions,
+            exclude_functions=db_settings.exclude_functions,
+            spatial=db_settings.only_spatial_tables,
+        )
+
+    @app.on_event("shutdown")
+    async def shutdown_event() -> None:
+        """Close database connection."""
+        await close_db_connection(app)
+
+    return app
```

Comment:
our tests suite is a bit complex because we want to test our main application (`tipg.main.app`) but also test some customization. Sadly we have a lot of things that get initialized on `import` when importing `tipg.main.app` so this little help tries to create an application similar (but enable customization)